newsroom/wire/search.py

```diff
@@ -215,7 +223,58 @@ def test_new_item(self, item_id, topics, users, companies):
                     topic_matches.append(topic['_id'])
 
         except Exception as exc:
-            logger.error('Error in test_new_item for query: {}'.format(json.dumps(source)),
+            logger.error('Error in get_matching_topics for query: {}'.format(json.dumps(source)),
                          exc, exc_info=True)
 
         return topic_matches
+
+    def get_items(self, item_ids):
+        try:
+            query = {
+                'bool': {
+                    'must_not': [
+                        {'term': {'type': 'composite'}},
+                        {'constant_score': {'filter': {'exists': {'field': 'nextversion'}}}},
+                        {'term': {'pubstatus': 'canceled'}}
+                    ],
+                    'must': [
+                        {'terms': {'_id': item_ids}}
+                    ],
+                }
+            }
+
+            source = {'query': query}
+            source['size'] = len(item_ids)
+
+            req = ParsedRequest()
+            req.args = {'source': json.dumps(source)}
+
+            return super().get(req, None)
+
+        except Exception as exc:
+            logger.error('Error in get_matching_bookmarks for query: {}'.format(json.dumps(source)),
+                         exc, exc_info=True)
+
+    def get_matching_bookmarks(self, item_ids, users, companies):
+        """
+        Returns a list of user ids bookmarked any of the given items
+        :param item_id: list of ids of items to be searched
+        :param users: user_id, user dictionary
+        :param companies: company_id, company dictionary
+        :return:
+        """
+        bookmark_users = []
+
+        search_results = self.get_items(item_ids)
+
+        if not search_results:
+            return bookmark_users
+
+        for result in search_results.hits['hits']['hits']:
+            bookmarks = result['_source'].get('bookmarks', [])
+            for bookmark in bookmarks:
+                user = users.get(bookmark)
+                if user and str(user.get('company', '')) in companies:
```

Comment:
not sure if this companies check is needed, this would make old bookmarks stop working if user joins/changes a company, not sure if that's required?

Reply:
This is to check if the company of the user is active or not. It will still work if user changes a company, as long as company is active. The name of companies should be active_companies as in this form it looks confusing.